### PR TITLE
Feat/new identities

### PR DESCRIPTION
**Pull request by JMounier**

### CODE REVIEW

File: packages/identity/README.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ The ID of a user in your application.
 <br /><br />
 
 ```javascript
-createProvisionalIdentity(appId, email);
+createProvisionalIdentity(appId, 'email', email);
 ```
 
 Create a Tanker provisional identity. It allows you to share a resource with a user who does not have an account in your application yet.
```
Comment (jeremad) [inline packages/identity/README.md:35]: Please add the new `upgradeIdentity`

Comment (tux3) [reply packages/identity/README.md:35]: Skipping per off-list discussion

**Review by jeremad — APPROVED** [review]

File: packages/identity/src/identity.js [hunk 2]
```diff
@@ -286,7 +296,11 @@ export async function getPublicIdentity(tankerIdentity: b64string): Promise<b64s
   }
 
   if (identity.public_signature_key && identity.public_encryption_key) {
-    const { trustchain_id, target, value, public_signature_key, public_encryption_key } = identity; // eslint-disable-line camelcase
+    const { trustchain_id, public_signature_key, public_encryption_key } = identity; // eslint-disable-line camelcase
+    let { target } = identity;
+    const value = await _getPublicHashedValueFromSecretProvisional(identity);
+    if (target === 'email') // For compatibility reasons, for hashed_email pub prov ids, the private is just 'email'
```
Comment (blastrock) [inline packages/identity/src/identity.js]: Not sure I understand this comment. It's the same for phone, there's no compatibility issue here, is it?

File: packages/identity/src/identity.js [hunk 3]
```diff
@@ -307,9 +307,17 @@ export async function getPublicIdentity(tankerIdentity: b64string): Promise<b64s
   throw new InvalidArgument(`Invalid secret identity provided: ${tankerIdentity}`);
 }
 
-export function upgradeIdentity(tankerIdentity: b64string): b64string {
+export async function upgradeIdentity(tankerIdentity: b64string): b64string {
   if (!tankerIdentity || typeof tankerIdentity !== 'string')
     throw new InvalidArgument('tankerIdentity', 'b64string', tankerIdentity);
-  const identity = _deserializeIdentity(tankerIdentity);
+  const frozenIdentity = _deserializeIdentity(tankerIdentity);
+  // $FlowIgnore flow doesn't understand that the spread doesn't change the type
```
Comment (blastrock) [inline packages/identity/src/identity.js:322]: We don't use flow anymore, do we?

File: packages/identity/src/identity.js [hunk 4]
```diff
@@ -6,8 +6,8 @@ import { obfuscateUserId } from './userId';
 import { createUserSecretB64 } from './userSecret';
 
 type PermanentIdentityTarget = 'user';
-type SecretProvisionalIdentityTarget = 'email';
-type PublicProvisionalIdentityTarget = 'email' | 'hashed_email';
+type SecretProvisionalIdentityTarget = 'email' | 'phone_number';
+type PublicProvisionalIdentityTarget = 'email' | 'hashed_email' | 'phone_number';
```
Comment (blastrock) [inline packages/identity/src/identity.js]: hashed_phone_number?

**Review by blastrock — COMMENTED** [review]
couldn't finish the review, brb

File: packages/identity/src/identity.js (hunk 2 above)
Comment (tux3) [reply packages/identity/src/identity.js]: There used to be, the comment is stale. Fixed.

File: packages/identity/src/identity.js (hunk 4 above)
Comment (tux3) [reply packages/identity/src/identity.js]: Sorry, again this is a feature branch so you are looking through the history =)  
This was a decision that was taken later (and the TEP updated), so it is done in another commit.

File: packages/identity/src/identity.js (hunk 3 above)
Comment (tux3) [reply packages/identity/src/identity.js:322]: On master, I still see the following in `package.json`:

`"proof": "yarn lint:js && yarn lint:flow && yarn test"`

I'm not aware of the status of the Typescript migration, but at least as of this feature branch we still need Flow to pass the CI lint, I believe

**Review by blastrock — APPROVED** [review]

File: packages/identity/src/identity.js [hunk 5]
```diff
@@ -305,11 +305,11 @@ export async function getPublicIdentity(tankerIdentity: b64string): Promise<b64s
 
   if (identity.public_signature_key && identity.public_encryption_key) {
```
[the remark this answers is not in the document]
Comment (tux3) [reply packages/identity/src/identity.js:306]: It both satisfies flow and enables the `throw new InvalidArgument` below.  
I can remove that to match the other languages more closely, if you want.

File: packages/identity/src/identity.js (hunk 3 above)
Comment (JMounier) [reply packages/identity/src/identity.js:322]: the typescript pr https://github.com/TankerHQ/identity-js/pull/6 is not merged yet. It stills need a little testing (ie. target -> browser version). I'll handle the conflicts once this is merged

File: packages/identity/src/identity.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (blastrock) [reply packages/identity/src/identity.js:306]: That, or just a comment saying that you know this is a provisional identity but that `if` helps flow understand

[the remark this answers is not in the document]
Comment (tux3) [reply packages/identity/src/identity.js:307]: Okay, added a comment

[the remark this answers is not in the document]
Comment (blastrock) [reply packages/identity/src/identity.js:307]: Awesome!

**Pull request merged**